src/views/PersonalHistory/components/PersonalHistoryForm/index.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { forwardRef, useRef, useImperativeHandle } from 'react';
+import { FormControl, Grid, Card } from '@material-ui/core';
+import {
+  cardInfoEtilismoItens,
+  cardInfoTabagismoItens,
+} from 'views/PersonalHistory/statics';
+import CardInfo from '../CardInfo';
+import { useHistory } from 'react-router-dom';
+import { useToast } from 'hooks/toast';
+import { usePatient } from 'context/PatientContext';
+import { Form, Formik } from 'formik';
+import GenericRadioGroup from 'components/Forms/GenericRadioGroup';
+import GenericCheckboxGroup from 'components/Forms/GenericCheckboxGroup';
+import api from 'services/api';
+import useStyles from './styles';
+import { getInitialValuesPersonalHistoryForm } from 'models/personalHistory/PersonalHistoryService';
+
+const PersonalHistoryForm = (props, ref) => {
+  const {
+    patientHistory,
+    drogas,
+    tiposSitucaoEtilismo,
+    tiposSitucaoTabagismo,
+    tiposSitucaoUsoDrogas,
+  } = props;
+
+  const classes = useStyles();
+
+  const history = useHistory();
+
+  const refFormik = useRef(null);
+
+  const { addToast } = useToast();
+
+  const { patient } = usePatient();
+
+  const initialValues = getInitialValuesPersonalHistoryForm(patientHistory);
+
+  const handleSubmit = async values => {
+    try {
+      // pegando as drogas seleciondas
+      const selectedDrogasIds = Object.entries(values.drogas).map(elem =>
+        elem[1] ? Number(elem[0]) : null,
+      );
+
+      // Sanitizando o values antes de enviar para a request
+      const valuesSanitized = Object.keys(values).reduce((acc, curr) => {
+        if (values[curr] !== '') {
+          return { ...acc, [curr]: values[curr] };
+        }
+        return acc;
+      }, {});
+
+      await api.post(`/pacientes/${patient.id}/historico`, {
+        ...valuesSanitized,
+        drogas: selectedDrogasIds,
+      });
+
+      addToast({
+        type: 'success',
+        message: 'Dados salvos com sucesso',
+      });
+      history.push('/categorias');
+    } catch (err) {
+      addToast({
+        type: 'error',
+        message: 'Erro ao tentar registrar história pessoal, tente novamente',
+      });
+    }
```

Comment:
Achei que tinha comentado essa parte.

A ideia do arquivo modelService é para remover essa parte lógica do componente, cuja responsabilidade é só visual.
Essa parte inteira pode ser enviada ao serviço, deixando aqui somente:
```javascript
try {
  await postPatientHistory(values)
  addToast({
    type: 'success',
    message: 'Dados salvos com sucesso',
  });
  history.push('/categorias');
} catch (err) {
  addToast({
    type: 'error',
    message: 'Erro ao tentar registrar história pessoal, tente novamente',
  });
}
```

E no serviço ficaria a lógica dos dados e o submit:

````javascript
      return api.post(`/pacientes/${patient.id}/historico`, {
        ...Object.keys(values).reduce((acc, curr) => {
          if (values[curr] !== '') {
            return { ...acc, [curr]: values[curr] };
          }
          return acc;
        }, {});,
        drogas: Object.entries(values.drogas).map(elem => elem[1] ? Number(elem[0]) : null),
      });
````

Se qualquer comentário no código for preciso para explicar um escopo, é melhor criar um método que identifique o escopo. Isso é bem do clean code :)

Reply:
Realmente @chicaothiago, separar isso deixa o código bem mais **clean** 👏🏻 👏🏻 👏🏻